### PR TITLE
Donation v1.2: release candidate

### PR DESCRIPTION
**Pull request by mahmed0715**
Fix https://github.com/Plant-for-the-Planet-org/treecounter-app/issues/1670

Repeated from issue:
- [x] In Single Projects Page, the price doesn't work property, so implement a floating button.
- [x] In project list view target, comma is missing on number less than 100k eg (100,000)
- [x] Fix Donation view, when support is being clicked, Instead of bring users to Donate trees without option to go back

### CODE REVIEW

**Comment by mahmed0715** [comment]
1. solved : price is now good cliable, Float the donate button

**Comment by mahmed0715** [comment]
2. solved. format target (i18n works)

**Comment by mahmed0715** [comment]
@sagararyal the back button from support button works now.

Note for developer: Added inner route app_supportTrees, so next time when we need to redirect donate from support button we need to use this inner route.

File: app/components/PublicTreeCounter/PublicTreecounter.native.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ class PublicTreeCounter extends React.Component {
 
   onRegisterSupporter() {
     this.props.supportTreecounterAction(this.props.treecounter);
-    updateRoute('app_donateTrees', this.props.navigation, 0, {
+    updateRoute('app_supportTrees', this.props.navigation, 44, {
```
Comment (norbertschuler) [inline app/components/PublicTreeCounter/PublicTreecounter.native.js]: Just as I am curious - how did you decide about using the number `44` as the id parameter?

File: app/styles/selectplantproject/selectplantproject-full.native.js [hunk 2]
```diff
@@ -53,6 +53,12 @@ export default EStyleSheet.create({
     flexShrink: 0,
     padding: 16
   },
+  floatingButtonContainer: {
+    padding: 16,
+    position: 'absolute',
+    bottom: 35,
+    right: 0
+  },
```
Comment (norbertschuler) [inline app/styles/selectplantproject/selectplantproject-full.native.js]: I am not a designer, but I would prefer a rounded button as @harshvitra used them for competitions or pledge events. Maybe he can share his styles with you. Also an short action button as `(+)` is commonly positioned at the bottom right edge of the screen while a bigger button with text on it as you designed it is more often centered at the bottom of the screen.

File: app/components/Competition/CompetitionParticipant.native.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ class CompetitionParticipant extends React.Component {
     };
     console.log('support button pressed');
     this.props.supportTreecounterAction(supportObject);
-    updateRoute('app_donateTrees', this.props.navigation, 0, {
+    updateRoute('app_supportTrees', this.props.navigation, 55, {
```
Comment (norbertschuler) [inline app/components/Competition/CompetitionParticipant.native.js]: Just as I am curious - how did you decide about using the number 55 as the id parameter? What will happen if we accidentally use the same number twice?

File: app/components/PublicTreeCounter/PublicTreecounter.native.js (hunk 1 above)
Comment (mahmed0715) [reply app/components/PublicTreeCounter/PublicTreecounter.native.js]: The code looks for only 0 value, so without 0 anything will work...

File: app/styles/selectplantproject/selectplantproject-full.native.js (hunk 2 above)
Comment (mahmed0715) [reply app/styles/selectplantproject/selectplantproject-full.native.js]: For this case we need donate text over there, I can center it, but @sagararyal suggested a button like we have on competition like the + button.. we can not use + for this case...
@sagararyal can you please check the current button

File: app/components/Competition/CompetitionParticipant.native.js (hunk 3 above)
Comment (mahmed0715) [reply app/components/Competition/CompetitionParticipant.native.js]: same as before, the logic checks for only 0 value, so any value other than 0 will work..

File: app/components/PublicTreeCounter/PublicTreecounter.native.js (hunk 1 above)
Comment (norbertschuler) [reply app/components/PublicTreeCounter/PublicTreecounter.native.js]: Ok, right, it does not seem to matter for https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/27deef1adee82362e233544a3be7def314ccb315/app/helpers/routerHelper/routerHelper.native.js#L11 I had been looking at the file for the web client where the id is used for a state at https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/27deef1adee82362e233544a3be7def314ccb315/app/helpers/routerHelper/routerHelper.js#L7

File: app/styles/selectplantproject/selectplantproject-full.native.js (hunk 2 above)
Comment (norbertschuler) [reply app/styles/selectplantproject/selectplantproject-full.native.js]: So maybe at least use rounded corners ;-)

Comment (sagararyal) [reply app/styles/selectplantproject/selectplantproject-full.native.js]: clarification I suggested button with rounded corners like in Edit/ Add Competition pages, not + button.

Comment (mahmed0715) [reply app/styles/selectplantproject/selectplantproject-full.native.js]: implemented new design only for bottom area on this PR

**Comment by mahmed0715** [comment]
@sagararyal please check the bottom area is implemented on this PR, we can merge, 
I will create another PR with the new design.

**Comment by mahmed0715** [comment]
Screenshot_1575868480
latest look

**Comment by sagararyal** [comment]
There is a known issue for iOS, but I am going to merge this and create a new issue for that.

**Pull request merged**